streamer/src/nonblocking/quic.rs

```diff
@@ -1864,11 +1860,13 @@ pub mod test {
             CONNECTION_CLOSE_CODE_DROPPED_ENTRY.into(),
             CONNECTION_CLOSE_REASON_DROPPED_ENTRY,
         );
-        // Wait long enough for the stream to timeout in receiving chunks
-        let sleep_time = Duration::from_secs(1).min(WAIT_FOR_STREAM_TIMEOUT * 1000);
-        sleep(sleep_time).await;
 
-        assert_eq!(stats.connection_removed.load(Ordering::Relaxed), 1);
+        let start = Instant::now();
+        while stats.connection_removed.load(Ordering::Relaxed) != 1 {
```

Comment:
Here and below, I made it look waiting for the expected condition. If the expected conditions are never reached, the test fails by timing out. I still put an upper bound (1s) on how long it's ok to wait for the server to close the connections. 1s should be enough to account for terrible scheduling perf in CI if things get overloaded.